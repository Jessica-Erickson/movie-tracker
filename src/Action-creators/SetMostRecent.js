export const setMostRecent = (movies) => ({
  type: 'SET_MOST_RECENT',
  movies: movies.results
})