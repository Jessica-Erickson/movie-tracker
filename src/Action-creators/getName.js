export const getName = (name) => ({
  type: 'GET_NAME',
  name
})