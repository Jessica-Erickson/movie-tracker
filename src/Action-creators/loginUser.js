export const loginUser = (userName) => ({
  type: 'LOGIN_USER',
  userName,
})