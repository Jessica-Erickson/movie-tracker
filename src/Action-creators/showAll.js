export const showAll = () => ({
  type: 'SHOW_ALL'
})