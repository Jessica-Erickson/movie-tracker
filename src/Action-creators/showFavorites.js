export const showFavorites = () => ({
  type: 'SHOW_FAVORITES'
})