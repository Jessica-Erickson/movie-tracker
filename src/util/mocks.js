import { key } from './key.js'

export const fetch = jest.fn().mockImplementation(() => Promise.resolve({
        ok: true,
        json: () => Promise.resolve(mock.mockResults)
      }))

export const mockResults = { 
  results: [
    { 
      poster_path: '2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg',
      title: 'Venom',
      vote_average: 5 ,
      overview: 'This movie is RAD!',
      release_date: '2018-9-11',
    }
  ] 
}

export const mockMovie = { 
      poster_path: '2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg',
      title: 'Venom',
      vote_average: 5 ,
      overview: 'This movie is RAD!',
      release_date: '2018-9-11',
    }

export const mockState = { 
    movies: [
      { 
        poster_path: '2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg',
        title: 'Venom',
        vote_average: 5 ,
        overview: 'This movie is RAD!',
        release_date: '2018-9-11',
      }
    ],
    filter: 'favorites' 
  };

export const url = `https://api.themoviedb.org/3/movie/now_playing?api_key=${key}&language=en-US&page=1`;

export const userResponse = {
  data: {
    id: 2,
    name: 'Jessica',
    password: 'passw0rd',
    email: 'jerick999@gmail.com',
  },
  message: 'Retrieved ONE User',
  status: 'success',
}

export const user = {
  password: 'passw0rd',
  email: 'jerick999@gmail.com',
}

export const options = {
  method: 'POST',
  body: JSON.stringify({
  password: 'passw0rd',
  email: 'jerick999@gmail.com',
    }),
  headers: {
    'Content-Type': 'application/json'
  },
}

export const newUserResponse = {
  id: 4,
  message: "New user created",
  status: "success",
}

export const newUser = {
  name: 'Tim', 
  password: 'm30w',
  email: 'tag.garrity@gmail.com', 
}

export const newOptions = {
  method: 'POST',
  body: JSON.stringify({
  name: 'Tim',
  password: 'm30w',
  email: 'tag.garrity@gmail.com',
    }),
  headers: {
    'Content-Type': 'application/json'
  },
}