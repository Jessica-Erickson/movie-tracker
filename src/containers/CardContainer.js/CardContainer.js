import React, { Component } from 'react';
import {connect} from 'react-redux';
import * as API from '../../util/API.js';
import {setMostRecent} from '../../Action-creators/SetMostRecent';


export class CardContainer extends Component{
  constructor() {
    super()
  }
  componentDidMount = async () => {
    const initialMovieData = await API.getMovieData()
    this.props.onload(initialMovieData)
  }
 render() {
  return (
    <div>
    </div>
  )
 } 
}


export const mapStateToProps = state => ({
  movies: state.movies
})

export const mapDispatchToProps = dispatch => ({
  onload: (movies) => dispatch(setMostRecent(movies))
})

export default connect(mapStateToProps, mapDispatchToProps)(CardContainer);