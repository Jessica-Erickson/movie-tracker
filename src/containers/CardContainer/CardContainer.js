import React, { Component } from 'react';
import {connect} from 'react-redux';
import {setMostRecent} from '../../Action-creators/setMostRecent';
import * as API from '../../util/helper';
import Card from '../Card/Card';
import './CardContainer.css'

export class CardContainer extends Component {
  componentDidMount = async () => {
    try {
      const initialMovieData = await API.getMovieData();
      this.props.onload(initialMovieData)
      return initialMovieData;
    } catch (e) {
      throw Error(e.message);
    }
  }

  render() {
    const { movies } = this.props;
    if(movies){
      const movieCards = movies.map(movie => {
        return <Card key={movie.id} {...movie} />
      })
      return (
        <section className='card-container'>
          {movieCards}
        </section>
      ) 
    } else {
      return null;
    }

  } 
}


export const mapStateToProps = state => ({
  movies: state.movies
})

export const mapDispatchToProps = dispatch => ({
  onload: (movies) => dispatch(setMostRecent(movies))
})

export default connect(mapStateToProps, mapDispatchToProps)(CardContainer);

