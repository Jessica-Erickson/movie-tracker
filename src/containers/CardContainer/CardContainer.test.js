import React from 'react'
import CardContainer from './CardContainer'
import {mapStateToProps, mapDispatchToProps} from './CardContainer'
import * as mocks from '../../util/mocks';
import {setMostRecent} from '../../Action-creators/SetMostRecent';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';

describe('CardContainer', () => {
  it('should match the snapshot', () => {
    const mockStore = configureMockStore()
    const initialState = {movies: []}
    const store = mockStore(initialState);

    const wrapper = shallow(<CardContainer store={store} />);

    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should return the expected object', () => {
      const expected = { movies: 
        [
          { 
            poster_path: '2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg',
            title: 'Venom',
            vote_average: 5 ,
            overview: 'This movie is RAD!',
            release_date: '2018-9-11',
          }
        ] 
      };

      expect(mapStateToProps(mocks.mockState)).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch when using a function in MDTP', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = setMostRecent(mocks.mockResults);
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.onload(mocks.mockResults);

      expect(mockDispatch).toBeCalledWith(actionToDispatch);
    });
  });
});
