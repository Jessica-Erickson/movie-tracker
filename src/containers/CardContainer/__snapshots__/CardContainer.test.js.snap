// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
<div>
  <Card
    id={1}
    key="1"
  />
  <Card
    id={2}
    key="2"
  />
  <Card
    id={3}
    key="3"
  />
</div>
`;
